### PR TITLE
[release-1.15] Update to go 1.12.7

### PR DESCRIPTION
**Pull request by tao12345666333**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature
> /kind flake

**What this PR does / why we need it**:

go1.12.7 (released 2019/07/08) includes fixes to cgo, the compiler,
and the linker.
See the <a href="https://github.com/golang/go/issues?q=milestone%3AGo1.12.7">Go 1.12.7 milestone</a> on our issue tracker for details.

https://github.com/golang/go/compare/go1.12.6...go1.12.7

**Special notes for your reviewer**:

This is part or #79912

**Does this PR introduce a user-facing change?**:

```release-note
update to use go 1.12.7
```

### CODE REVIEW

**Comment by tao12345666333** [comment]
Split commits so that we can cherry pick to release-1.14

**Comment by neolit123** [comment]
commented here:
https://github.com/kubernetes/kubernetes/issues/79912#issuecomment-511232632

File: cmd/kubeadm/app/phases/upgrade/compute_test.go [hunk 1]
```diff
@@ -892,7 +892,7 @@ func TestGetBranchFromVersion(t *testing.T) {
 			expectedVersion: "1.11",
 		},
 		{
-			version:         "1.12.5",
+			version:         "1.12.7",
```
Comment (cblecker) [inline cmd/kubeadm/app/phases/upgrade/compute_test.go]: I think this is incorrect.

File: staging/publishing/rules.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ rules:
       branch: release-1.15
       dir: staging/src/k8s.io/code-generator
     name: release-1.15
-    go: 1.12.5
+    go: 1.12.7
```
Comment (cblecker) [inline staging/publishing/rules.yaml]: are these changes needed @nikhita @sttts ? I don't think we've done these before when bumping go versions

File: cmd/kubeadm/app/phases/upgrade/compute_test.go (hunk 1 above)
Comment (tao12345666333) [reply cmd/kubeadm/app/phases/upgrade/compute_test.go]: It's my wrong. Thanks.

File: staging/publishing/rules.yaml (hunk 2 above)
Comment (nikhita) [reply staging/publishing/rules.yaml]: Yes, these are needed if we are bumping the version in the `release-1.15` branch.

> I don't think we've done these before when bumping go versions

We should have done that. :) Created https://github.com/kubernetes/kubernetes/pull/80151 and https://github.com/kubernetes/publishing-bot/pull/197. 

It doesn't _really_ matter because the go version here is mainly used while testing client-go and there haven't been any particular changes in client-go that are specifically related to features in new go versions, but it's definitely better to bump the version in case there are any bug fixes, etc. :+1:

File: cmd/kubeadm/app/phases/upgrade/compute_test.go (hunk 1 above)
Comment (tao12345666333) [reply cmd/kubeadm/app/phases/upgrade/compute_test.go]: @cblecker After I checked the code, the changes in this place didn't actually have any real impact.
It's just checking, and this part of the change doesn't change the final result.

Comment (tao12345666333) [reply cmd/kubeadm/app/phases/upgrade/compute_test.go]: Do I need to modify it back?

**Comment by cblecker** [comment]
Thanks @nikhita !

/ok-to-test

File: cmd/kubeadm/app/phases/upgrade/compute_test.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/phases/upgrade/compute_test.go]: @tao12345666333 please remove this change.
1.12.5 here did not mean go version.

Comment (tao12345666333) [reply cmd/kubeadm/app/phases/upgrade/compute_test.go]: @neolit123 Thanks. Addressed.

**Comment by tao12345666333** [comment]
/test pull-kubernetes-kubemark-e2e-gce-big

not sure if the test has been fixed.

**Comment by tao12345666333** [comment]
@cblecker PTAL. :smile_cat:

**Review by nikhita — APPROVED** [review]
/approve
/retest

**Comment by tpepper** [comment]
I'm going to declare this is 
/kind bug
because we are picking up key toolchain bugfixes.
